### ISSUE TITLE
Pseudo-label satellite imagery using an ML model. Then predict future PV using those labels

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/power_perceiver#178]
We've known for a little while that an ML model can do surprisingly well at inferring PV power at time _t_ from satellite imagery at time _t_. (Not predicting the future)

In OpenAI's new paper on "[learning to play minecraft with video pre-training](https://openai.com/blog/vpt/)", the train a model to infer actions from minecraft videos. This model is a 3D CNN, which feeds a transformer. Crucially, this model sees the _future_. Which makes it much more reliable at inferring actions. They then use this model to pseudo-label tens of thousands of hours of minecraft videos from YouTube.

Maybe we could do something similar. We train a model (perhaps a 3D CNN feeding a transformer) to infer PV power from satellite imagery (and NWPs?). We use this to pseudo-label satellite imagery for which we have no PV power data. We then train a similar-architecture model to predict future _PV_ (not future imagery) using all this data, in a supervised fashion.

Questions:

* How to train a model to _also_ make use of real PV data? Maybe that's a separate model component that combines the output of the "satellite inference model" with read PV obs?

Some notes from the paper:

* The 3D CNN is really important. They show its importance in the paper.

**Comment by jacobbieker** [comment]
Really interesting paper! For predicting the PV for whole region, I think the downsampling is a pretty good approach, that's how MetNet models incorporate huge context areas as well.

 I do really like the idea of making pseudo labels for the entire satellite area and all the extra training data that could give us. Maybe we could also get some PV data from Germany, or Italy so have extra "real" PV in more geographic areas for the labelling model to learn better? Although I guess it should be able to learn enough from just UK systems. 

Do the MetOffice NWPs extend across Europe? OR would we need to get other NWPs for those areas?

I would think directly predicting the real PV power would be more informative for the labeller than using processed data, if we want the labeller to match the real data as closely as possible. 

For the Questions:
I'd think we would want to save the psuedo labels to disk as they should be small like the current PV power data, and to ensure the labels are all the same for experiments. 
For inference, I think if the labelling model is good enough to provide realistic labels, then it makes sense that that would help at inference time as well, especially for GSPs with sparse PV data, or if PV data goes offline.

**Comment by jacobbieker** [comment]
These people are releasing a dataset on global GHI, DNI, and something else with ground truth set around the world: https://twitter.com/IEA_SolarPACES/status/1531884649013813249 not sure the timeline for it, but seemed to be soonish. 

I would think, just to start, creating a labeller for PV output might be simpler, because of the ground truth we have. Although both types would probably be quite useful! And yeah, I think having a set of datasets for different  types of PV systems, so then people can see how different types of PV systems would work in a location.

**Comment by akanshasingh803** [comment]
Have a look at our paper titled, "A Moment in the Sun: Solar Nowcasting from Multispectral Satellite Data using Self-Supervised Learning" where we have trained a global model using self-supervised learning to predict future satellite observations at t+1 using abundantly available unlabeled satellite data and further used them to nowcast solar 15 minutes into the future using another local solar model that takes into account historical solar generation as well as temperature values. Here are the links- https://dl.acm.org/doi/10.1145/3538637.3538854; https://arxiv.org/abs/2112.13974

**Comment by JackKelly** [comment]
That's really interesting to know, thank you @JamieTaylor-TUOS!

**Comment by JackKelly** [comment]
@jacobbieker while I've been off sick, I've been thinking a bit more about this approach, and I've updated the post at the top (including a new diagram!) The approach is still the same... hopefully I've answered a problem that was bugging me (about how to handle the difference between different PV systems!)

**Comment by JackKelly** [comment]
For the pseudo-labelling model, maybe also give it ground-truth PV data from neighbouring PV systems. But also train with lots of examples with no PV data (because, for example, there's no PV data over the ocean). So, maybe, for a given ROI, if there's more than 1 PV system then use a random proportion (but always at least 1 PV system) as the target, and the rest as inputs. But frequently drop out all the PV inputs.
